Source/Scene/Points3DTileContent.js

```diff
@@ -164,159 +196,366 @@ define([
-            this._constantColor = Color.DARKGRAY;
+        if (!hasColors && !isConstantColor) {
+            // Use a default constant color
+            content._constantColor = Color.clone(Color.DARKGRAY, content._constantColor);
         }
 
-        var vs = 'attribute vec3 position3DHigh; \n' +
-                 'attribute vec3 position3DLow; \n' +
-                 'uniform float pointSize; \n';
+        var vs = 'attribute vec3 a_position; \n' +
+                 'varying vec4 v_color; \n' +
+                 'uniform float u_pointSize; \n' +
+                 'uniform vec4 u_highlightColor; \n';
+
         if (hasColors) {
-            if (translucent) {
-                vs += 'attribute vec4 color; \n' +
-                      'varying vec4 v_color; \n';
+            if (isTranslucent) {
+                vs += 'attribute vec4 a_color; \n';
             } else {
-                vs += 'attribute vec3 color; \n' +
-                      'varying vec3 v_color; \n';
+                vs += 'attribute vec3 a_color; \n';
             }
         }
-        vs += 'void main() \n' +
-              '{ \n' +
-              '    gl_Position = czm_modelViewProjectionRelativeToEye * czm_computePosition(); \n' +
-              '    gl_PointSize = pointSize; \n';
-        if (hasColors) {
-            vs += '    v_color = color; \n';
+        if (hasNormals) {
+            if (isOctEncoded16P) {
+                vs += 'attribute vec2 a_normal; \n';
+            } else {
+                vs += 'attribute vec3 a_normal; \n';
+            }
         }
-        vs += '}';
 
-        var fs = 'uniform vec4 highlightColor; \n';
+        if (isQuantized) {
+            vs += 'uniform vec3 u_quantizedVolumeScale; \n';
+        }
+
+        vs += 'void main() \n' +
+              '{ \n';
+
         if (hasColors) {
-            if (translucent) {
-                fs += 'varying vec4 v_color; \n';
+            if (isTranslucent) {
+                vs += '    vec4 color = a_color * u_highlightColor; \n';
             } else {
-                fs += 'varying vec3 v_color; \n';
+                vs += '    vec4 color =  vec4(a_color * u_highlightColor.rgb, u_highlightColor.a); \n';
             }
+        } else {
+            vs += '    vec4 color = u_highlightColor; \n';
         }
-        fs += 'void main() \n' +
-              '{ \n';
 
-        if (hasColors) {
-            if (translucent) {
-                fs += '    gl_FragColor = v_color * highlightColor; \n';
+        if (hasNormals) {
+            if (isOctEncoded16P) {
+                vs += '    vec3 normal = czm_octDecode(a_normal); \n';
             } else {
-                fs += '    gl_FragColor = vec4(v_color * highlightColor.rgb, highlightColor.a); \n';
+                vs += '    vec3 normal = a_normal; \n';
             }
+
+            vs += '    normal = czm_normal * normal; \n' +
+                  '    color *= max(dot(normal, czm_sunDirectionEC), 0.0); \n';
```

Comment:
I never really thought beyond "per-point normals for lighting."  Are you sure that all we want to do for now is this diffuse term?  Not the Cesium built-in GLSL functions?  We can also revisit this soon when we do point cloud styling.